### PR TITLE
GH-12190: Remove checkout dependencies in PromotionService

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-12190

### CODE REVIEW

File: projects/storefrontlib/src/shared/services/promotion/abstract-promotion.service.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { Cart, Order, OrderEntry, PromotionResult } from '@spartacus/core';
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/shared/services/promotion/abstract-promotion.service.ts:1]: How about this:
```
@Injectable()
export abstract class PromotionService {
  protected loacatin: PromotionLocation;
  
  abstract getOrderPromotions();

  // then the functions used by children
  getProductPromotionForEntry;
  getProductPromotion;
  isConsumedByEntry;
```
Then in checkout:
```
@Injectable({
  providedIn: 'root',
})
export class CheckoutPromotionService extends PromotionService {
  location = PromotionLocation.Checkout;

  // then implement the getOrderPromotions(), also move the getOrderPromotionsFromOrderHelper to this class.
```
When use it in checkout:
```
providers: [
    {
       provide: PromotionService,
       useExisting: CheckoutPromotionService,
    },
],
```
same thing for the `Order` and `ActiveCart`, then we can use `PromotionService` everywhere.

Comment (WeizhengSap) [reply projects/storefrontlib/src/shared/services/promotion/abstract-promotion.service.ts:1]: `getOrderPromotion` maybe can be renamed to `getPromotion`?

**Review by WeizhengSap — CHANGES_REQUESTED** [review]
please check my suggestion.

**Comment by plabadie** [comment]
Replaced by #12686

**Pull request closed without merging**